### PR TITLE
Модуль Основы JS. Задача 1

### PR DESCRIPTION
**Pull request by TatyanaSlyandina67**
Мои тесты выдали ошибку: 
C:\Users\slyte\20240711_project\js-20240711_severstal-slyte>npm test
 PASS  02-javascript-data-types/2-pick/index.spec.js
  javascript-data-types/pick
    √ should return new object with one passed field (3ms)
    √ should return new object with a few passed field
    √ should return an empty object if passed fields does not found (1ms)

Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 PASS  01-intro/1-sum/index.test.js
  intro/sum
    √ should return sum of numbers (2ms)

 FAIL  02-javascript-data-types/1-sort-strings/index.spec.js
  javascript-data-types/sort-strings
    × should return sorted by "asc" array of strings (4ms)
    √ should return a new sorted array (1ms)
    × should return sorted by "desc" array of strings in "en" locale (1ms)
    × should correctly sort language-specific characters in "ru" locale (1ms)
    × should correctly sort strings started from uppercase (1ms)
    × should correctly sort strings for mixed "en" and "ru" locales

  ● javascript-data-types/sort-strings › should return sorted by "asc" array of strings

    expect(received).toEqual(expected) // deep equality

    Expected: ["a", "b", "c"]
    Received: undefined

      3 | describe('javascript-data-types/sort-strings', () => {
      4 |   it('should return sorted by "asc" array of strings', () => {
    > 5 |     expect(sortStrings(['b', 'c', 'a'])).toEqual(['a', 'b', 'c']);
        |                                          ^
      6 |   });
      7 |
      8 |   it('should return a new sorted array', () => {

      at Object.<anonymous> (02-javascript-data-types/1-sort-strings/index.spec.js:5:42)

  ● javascript-data-types/sort-strings › should return sorted by "desc" array of strings in "en" locale

    expect(received).toEqual(expected) // deep equality

    Expected: ["c", "b", "a"]
    Received: undefined

      16 |
      17 |   it('should return sorted by "desc" array of strings in "en" locale', () => {
    > 18 |     expect(sortStrings(['b', 'c', 'a'], 'desc')).toEqual(['c', 'b', 'a']);
         |                                                  ^
      19 |   });
      20 |
      21 |   it('should correctly sort language-specific characters in "ru" locale', () => {

      at Object.<anonymous> (02-javascript-data-types/1-sort-strings/index.spec.js:18:50)

  ● javascript-data-types/sort-strings › should correctly sort language-specific characters in "ru" locale

    expect(received).toEqual(expected) // deep equality

    Expected: ["абрикос", "ёжик", "яблоко"]
    Received: undefined

      20 |
      21 |   it('should correctly sort language-specific characters in "ru" locale', () => {
    > 22 |     expect(sortStrings(['абрикос', 'яблоко', 'ёжик'])).toEqual(['абрикос', 'ёжик', 'яблоко']);
         |                                                        ^
      23 |   });
      24 |
      25 |   it('should correctly sort strings started from uppercase', () => {

      at Object.<anonymous> (02-javascript-data-types/1-sort-strings/index.spec.js:22:56)

  ● javascript-data-types/sort-strings › should correctly sort strings started from uppercase

    expect(received).toEqual(expected) // deep equality

    Expected: ["Абрикос", "абрикос", "Ёжик", "ёжик", "Яблоко", "яблоко"]
    Received: undefined

      25 |   it('should correctly sort strings started from uppercase', () => {
      26 |     expect(sortStrings(['абрикос', 'Абрикос', 'яблоко', 'Яблоко', 'ёжик', 'Ёжик']))
    > 27 |       .toEqual(['Абрикос', 'абрикос', 'Ёжик', 'ёжик', 'Яблоко', 'яблоко']);
         |        ^
      28 |
      29 |     expect(sortStrings(['apple', 'Apple', 'banana', 'Banana', 'orange', 'Orange']))
      30 |       .toEqual(['Apple', 'apple', 'Banana', 'banana', 'Orange', 'orange']);

      at Object.<anonymous> (02-javascript-data-types/1-sort-strings/index.spec.js:27:8)

  ● javascript-data-types/sort-strings › should correctly sort strings for mixed "en" and "ru" locales

    expect(received).toEqual(expected) // deep equality

    Expected: ["Детский велосипед Lexus Trike Racer Trike", "Соска (пустышка) NUK 10729357", "Соска (пустышка) Philips SCF182/12", "ТВ тюнер D-COLOR  DC1301HD", "Powerbank аккумулятор Hiper SP20000"]
    Received: undefined

      56 |     ];
      57 |
    > 58 |     expect(sortStrings(data, 'asc')).toEqual(expectedAsc);
         |                                      ^
      59 |     expect(sortStrings(data, 'desc')).toEqual(expectedDesc);
      60 |   });
      61 | });

      at Object.<anonymous> (02-javascript-data-types/1-sort-strings/index.spec.js:58:38)

 PASS  02-javascript-data-types/3-omit/index.spec.js
  javascript-data-types/omit
    √ should return a new object without passed field
    √ should return a new object without few passed fields (1ms)
    √ should return initial object if passed fields doesn't found

Test Suites: 1 failed, 3 passed, 4 total
Tests:       5 failed, 8 passed, 13 total
Snapshots:   0 total
Time:        2.937s
Ran all test suites.

### CODE REVIEW

**Comment by jsru-1** [comment]
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

**Comment by jsru-1** [comment]
Добавляю преподавателя (@ufocoder) для код-ревью.

File: 02-javascript-data-types/1-sort-strings/index.js [hunk 1]
```diff
@@ -4,6 +4,30 @@
  * @param {string} [param="asc"] param - the sorting type "asc" or "desc"
  * @returns {string[]}
  */
-export function sortStrings(arr, param = 'asc') {
+export function sortStrings(arr, param='asc' ) {
+        let arrNew = [...arr];
```
Comment (ufocoder) [inline 02-javascript-data-types/1-sort-strings/index.js:8]: `let` -> `const`

File: 02-javascript-data-types/1-sort-strings/index.js [hunk 2]
```diff
@@ -4,6 +4,30 @@
...
+        return arrNew.sort((a, b) => {
+          if (param === 'asc') {
+            return a.localeCompare(b, ["ru", "en"], {caseFirst: 'upper'});
+          } 
+          if (param === 'desc') {
+            return b.localeCompare(a, ["ru", "en"], {caseFirst: 'lower'});
+          }
+        })
 
+        /*
```
Comment (ufocoder) [inline 02-javascript-data-types/1-sort-strings/index.js:18]: Не отправляйте закомментированный код

**Review by ufocoder — APPROVED** [review]
Отправьте исправления в новом PR

**Pull request merged**